grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/query/ReasonerQueryImpl.java

```diff
@@ -178,7 +175,7 @@ private boolean isTransitive() {
     }
 
     boolean isAtomic() {
-        return selectAtoms().size() == 1;
+        return getAtoms().stream().filter(Atomic::isSelectable).count() == 1;
```

Comment:
For big queries yes. However `getAtoms()` just returns a member variable and the `isSelectable` check is cheap. I think that we should have the atomicity recognition at the pattern, not atom level at some point.